### ISSUE TITLE
[release] some questions about release flow

### ISSUE THREAD

**Issue by chenrui333** [superfly/flyctl#1915]
It looks like there is some release flow issue as some commit got included into three releases (for example [this commit](https://github.com/superfly/flyctl/commit/c6b3ea94db3388c04fe886fce8f9a5c8bdc5dfd5) got included into 489, 491 492 release notes), also cause some stress in the homebrew side:

- https://github.com/Homebrew/homebrew-core/pull/125879
- https://github.com/Homebrew/homebrew-core/pull/125880
- https://github.com/Homebrew/homebrew-core/pull/125881

**Comment by chenrui333** [comment]
np, but just to clarify the flow a bit.

**Comment by chenrui333** [comment]
I still find the release notes is not very user friendly, I wonder if there would be some future improvements for it.

**Comment by kzys** [comment]
Yeah. Our release notes are directly generated from commit messages and our commit messages are sometimes not really descriptive.

**Comment by redjonzaci** [comment]
@kzys being the last one to comment 😂, do we still need this issue?

**Comment by tomr-a5** [comment]
Thanks for reporting this. The duplicate release contents were traced to cutting a release from a feature branch, and #1916 added a guard for that case. The current release script now refuses normal releases outside `master` or `main`, so I’m going to close this as fixed. Please reopen if the current release flow can still produce the same duplicate releases.
